### PR TITLE
runtime-c-api: not installing cranelift if not necessary

### PR DESCRIPTION
**Pull request by YaronWittenstein**
having `default-features = false` lets us install the `runtime-c-api` with `default-backend-singlepass` without installing also `cranelift`

### CODE REVIEW

**Comment by Hywan** [comment]
I'm not sure we want Singlepass as the default backend for the C API.
I'm OK though to be able to disable it when selecting another backend.

**Comment by YaronWittenstein** [comment]
> I'm not sure we want Singlepass as the default backend for the C API.
> I'm OK though to be able to disable it when selecting another backend.

@Hywan 
The `singlepass` isn't the default
My bad, I've used bad naming here.

File: lib/runtime-c-api/Cargo.toml [hunk 1]
```diff
@@ -15,16 +15,21 @@ crate-type = ["cdylib", "rlib", "staticlib"]
...
 
 [dependencies.wasmer-runtime-core]
+default-features = false
 path = "../runtime-core"
 version = "0.6.0"
 
 [features]
+default = ["clif"]
 debug = ["wasmer-runtime/debug"]
+clif = ["wasmer-runtime/cranelift", "wasmer-runtime/default-backend-cranelift"]
```
Comment (Hywan) [inline lib/runtime-c-api/Cargo.toml]: Can you rename `clif` to `cranelift-backend` please?

File: lib/runtime-c-api/Cargo.toml [hunk 2]
```diff
@@ -15,16 +15,21 @@ crate-type = ["cdylib", "rlib", "staticlib"]
...
 llvm = ["wasmer-runtime/llvm"]
```
Comment (Hywan) [inline lib/runtime-c-api/Cargo.toml]: Can you rename `llvm` to `llvm-backend` please?

File: lib/runtime-c-api/Cargo.toml [hunk 3]
```diff
@@ -15,16 +15,21 @@ crate-type = ["cdylib", "rlib", "staticlib"]
...
+singlepass = ["wasmer-runtime/default-backend-singlepass", "wasmer-runtime-core/backend-singlepass"]
```
Comment (Hywan) [inline lib/runtime-c-api/Cargo.toml]: Can you rename `singlepass` to `singlepass-backend` please?

File: lib/runtime-c-api/Cargo.toml [hunk 4]
```diff
@@ -15,16 +15,21 @@ crate-type = ["cdylib", "rlib", "staticlib"]
...
 path = "../runtime-core"
 version = "0.6.0"
 
 [features]
+default = ["cranelift-backend"]
 debug = ["wasmer-runtime/debug"]
-llvm = ["wasmer-runtime/llvm"]
+cranelift-backend = ["wasmer-runtime/cranelift", "wasmer-runtime/default-backend-cranelift"]
+llvm-backend = ["wasmer-runtime/llvm"]
+singlepass-backend = ["wasmer-runtime/default-backend-singlepass", "wasmer-runtime-core/backend-singlepass"]
```
Comment (Hywan) [inline lib/runtime-c-api/Cargo.toml]: I think this is:

```diff
- wasmer-runtime-core/backend-singlepass
+ wasmer-runtime/backend-singlepass
```

Thoughts?

Comment (YaronWittenstein) [reply lib/runtime-c-api/Cargo.toml]: checking shortly

Comment (YaronWittenstein) [reply lib/runtime-c-api/Cargo.toml]: @Hywan I seem it should stay `wasmer-runtime/default-backend-singlepass`

https://github.com/wasmerio/wasmer/blob/master/lib/runtime/Cargo.toml#L41

Comment (Hywan) [reply lib/runtime-c-api/Cargo.toml]: I'm talking about the second item of the list. You are using  `wasmer-runtime-core/…`, I think it is `wasmer-runtime/…`.

File: lib/runtime-c-api/Cargo.toml [hunk 5]
```diff
@@ -15,16 +15,21 @@ crate-type = ["cdylib", "rlib", "staticlib"]
...
+default-features = false
 path = "../runtime-core"
 version = "0.6.0"
 
 [features]
+default = ["cranelift-backend"]
 debug = ["wasmer-runtime/debug"]
-llvm = ["wasmer-runtime/llvm"]
+cranelift-backend = ["wasmer-runtime/cranelift", "wasmer-runtime/default-backend-cranelift"]
+llvm-backend = ["wasmer-runtime/llvm"]
```
Comment (Hywan) [inline lib/runtime-c-api/Cargo.toml]: Why not adding `default-backend-llvm` to the list of features to enable with `llvm-backend`?

**Pull request merged**